packages/legend-query/src/components/QueryBuilderFunctionPanel.tsx

```diff
@@ -0,0 +1,653 @@
+            'query-builder-function-explorer-tree__icon': !fromDependency,
+            'query-builder-function-explorer-tree__dependency-icon':
+              fromDependency,
+          })}
+        />
+      )
+    ) : (
+      <PURE_FunctionIcon />
+    );
+    const selectNode = (): void => onNodeSelect?.(node);
+    const [, dragConnector, dragPreviewConnector] = useDrag(
+      () => ({
+        type:
+          node.packageableElement instanceof ConcreteFunctionDefinition
+            ? QUERY_BUILDER_FUNCTION_TREE_DND_TYPE.FUNCTION
+            : QUERY_BUILDER_FUNCTION_TREE_DND_TYPE.PACKAGE,
+        item: (): { node?: QueryBuilderPackageElementTreeNodeData } =>
+          node.packageableElement instanceof ConcreteFunctionDefinition
+            ? { node }
+            : {},
+      }),
+      [node],
+    );
+    // hide default HTML5 preview image
+    useEffect(() => {
+      dragPreviewConnector(getEmptyImage(), { captureDraggingState: true });
+    }, [dragPreviewConnector]);
+
+    return (
+      <div>
+        <div
+          className="tree-view__node__container query-builder-function-explorer-tree__node__container"
+          onClick={selectNode}
+          ref={!isExpandable ? dragConnector : undefined}
+          style={{
+            paddingLeft: `${(level - 1) * (stepPaddingInRem ?? 1) + 0.5}rem`,
+            display: 'flex',
+          }}
+        >
+          {
+            <>
+              <div className="tree-view__node__icon query-builder-function-explorer-tree__node__icon">
+                <div className="query-builder-function-explorer-tree__expand-icon">
+                  {nodeExpandIcon}
+                </div>
+                <div
+                  className={'query-builder-function-explorer-tree__type-icon'}
+                >
+                  {nodeTypeIcon}
+                </div>
+              </div>
+              <div className="tree-view__node__label query-builder-function-explorer-tree__node__label--with-action">
+                {name}
+              </div>
+              <div className="query-builder-function-explorer-tree__node__actions">
+                {node.packageableElement instanceof
+                  ConcreteFunctionDefinition && (
+                  <QueryBuilderFunctionInfoTooltip
+                    element={node.packageableElement}
+                  >
+                    <div className="query-builder-function-explorer-tree__node__action query-builder-function-explorer-tree__node__info">
+                      <InfoCircleIcon />
+                    </div>
+                  </QueryBuilderFunctionInfoTooltip>
+                )}
+              </div>
+            </>
+          }
+        </div>
+      </div>
+    );
+  },
+);
+
+const QueryBuilderFunctionExplorerTreeNodeView = observer(
+  (
+    props: TreeNodeViewProps<
+      QueryBuilderPackageElementTreeNodeData,
+      {
+        queryBuilderState: QueryBuilderState;
+        fromDependency: boolean;
```

Comment:
probably don't need this right? we can follow what we do in the project explorer tree.